### PR TITLE
Various message component bug fixes

### PR DESCRIPTION
**Pull request by jr1221**
# Description

Fixes issue where the structure of a modal is created incorrectly.  A draft PR as modal improvements seem to still be needed.

Edit: second commit fixes modal interaction.

## Connected issues & other potential problems
Modals still cannot be recieved, as multiple exceptions are triggered.

Edit: above fixed.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] Ran `dart analyze` or `make analyze` and fixed all issues
- [x] Ran `dart format --set-exit-if-changed -l 160 ./lib` or `make format` and fixed all issues
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by jr1221** [comment]
I am just collecting various hacky fixes here.  I do not intended to actually ask for this to be merged unless @abitofevrything wants me to, because I am sure he has better solutions to these problems.

File: lib/src/http/managers/message_manager.dart
Comment (Rapougnac) [inline lib/src/http/managers/message_manager.dart:238]:
```suggestion
          style: maybeParse(raw['style'], TextInputStyle.parse),
```

**Review by Rapougnac — CHANGES_REQUESTED** [review]

**Comment by jr1221** [comment]
Alright I think this is ready to go.

**Review by l7ssha — APPROVED** [review]

**Comment by jr1221** [comment]
Sorry if I messed up the review process very new to public repositories.

File: lib/src/builders/interaction_response.dart
Comment (abitofevrything) [inline lib/src/builders/interaction_response.dart:133]: Could you change `components` from `List<TextInputBuilder>` to a `List<ActionRowBuilder>`, and then:
```suggestion
        'components': components.map((e) => e.build()).toList(),
```
This would avoid us having to make a breaking change if Discord ever adds support for different components in modals.

**Review by abitofevrything — CHANGES_REQUESTED** [review]
Looking good, just one small change for some future proofing.

> Sorry if I messed up the review process very new to public repositories.

No worries, everything's fine! I just asked to review this last night, which is why it didn't get merged right after l7ssha'a approval.

**Review by Rapougnac — APPROVED** [review]

File: lib/src/builders/interaction_response.dart
Comment (jr1221) [reply lib/src/builders/interaction_response.dart:133]: Wouldn't that break `nyxx-commands` as it is expecting to send `List<TextInputBuilder>` to `ModalBuilder`?

Comment (abitofevrything) [reply lib/src/builders/interaction_response.dart:133]: nyxx_commands can always be updated

Comment (l7ssha) [reply lib/src/builders/interaction_response.dart:133]: PR is targeting 6.0 so we can intruduce breaking changes

**Pull request merged**
